Automattic/jetpack
# Add lint configs for VS Code

## Proposed changes:

This adds VS Code configuration for PHPCS and Prettier so their respective extensions use the correct executables inside the monorepo when active.

Extensions 



### Other information:


## Jetpack product discussion



pf4qpu-3P-p2

## Does this pull request change what data or activity we track or use?



N/A

## Testing instructions:






You'll need to have the following extensions installed:

- [PHPCS](https://marketplace.visualstudio.com/items?itemName=shevaua.phpcs)
- [Prettier](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode)

It might also be helpful to install the [ESLint extension](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint) even though that one seems to work well out of the box.

- Open any PHP file inside the monorepo and make some invalid changes to the syntax. The errors should be highlighted and you should see the related explanation from PHPCS when hovering over them.
- Open any JS file from inside the monorepo and make invalid syntax changes too. If you have the ESLint extension, it'll highlight the errors.
- In any JS file, you should also be able to run the `Format Document` which will run prettier for just that file, or your selection.

Note: prettier should be using our `wp-prettier` fork installed in the monorepo and document should be formatted according to WP coding standards, not the prettier default.
